File: pubsub/src/main/java/glitch/pubsub/object/adapters/MessageTypeAdapter.java
package glitch.pubsub.object.adapters;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import glitch.pubsub.object.enums.MessageType;

import java.io.IOException;

public class MessageTypeAdapter extends TypeAdapter<MessageType> {
    @Override
    public void write(JsonWriter out, MessageType value) throws IOException {
        out.value(value.name());
    }

    @Override
    public MessageType read(JsonReader in) throws IOException {
        return MessageType.valueOf(in.nextString());
    }
}

File: pubsub/src/main/kotlin/io/glitchlib/pubsub/model/adapters/MessageTypeAdapter.kt
package io.glitchlib.pubsub.model.adapters

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import io.glitchlib.pubsub.model.MessageType
import java.io.IOException

class MessageTypeAdapter : TypeAdapter<MessageType>() {
    @Throws(IOException::class)
    override fun write(out: JsonWriter, value: MessageType) {
        out.value(value.name)
    }

    @Throws(IOException::class)
    override fun read(`in`: JsonReader): MessageType {
        return MessageType.valueOf(`in`.nextString())
    }
}
